### PR TITLE
Enhance Integration Tests

### PR DESCRIPTION
**Pull request by FelixRottler**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

- use maven verify instead of a docker-compose.yml for starting integration tests on the build pipeline
- run distribution integration tests with an integrated cloudserver container

### CODE REVIEW

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/integration/BaseS3IntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.junit.jupiter.api.BeforeAll;
+
+public abstract class BaseS3IntegrationTest {
+
+  @BeforeAll
+  public static void setupBucket() {
+    AWSCredentials credentials = new BasicAWSCredentials("accessKey1",
+        "verySecretKey1");
+
+    // Create a client connection based on credentials
```
Comment (mibrasap) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/integration/BaseS3IntegrationTest.java:18]: Might make sense to extract the commented blocks into dedicated methods

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/integration/S3PublisherTestIT.java [hunk 2]
```diff
@@ -47,6 +47,17 @@
   @Autowired
   private S3Publisher s3Publisher;
 
+
+  @BeforeEach
+  public void setup() {
+    objectStoreAccess.deleteObjectsWithPrefix("");
```
Comment (mibrasap) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/integration/S3PublisherTestIT.java:53]: Is there a specific method to "delete all"?

File: services/distribution/src/test/resources/application.yaml [hunk 3]
```diff
@@ -90,6 +90,14 @@ spring:
...
-        ddl-auto: validate
+    mixed: true
+  datasource:
+    driver-class-name: org.testcontainers.jdbc.ContainerDatabaseDriver
+    url: jdbc:tc:postgresql:11.8:///cwa?TC_TMPFS=/testtmpfs:rw?TC_INITSCRIPT=file:src/test/java/V0__init_db.sql
+  test:
+    database:
+      # Use datasource as defined above.
+      replace: none
+  jpa:
```
Comment (mibrasap) [inline services/distribution/src/test/resources/application.yaml:101]: Do we need it still? We switched from JPA to Spring Data JDBC

File: services/upload/src/test/java/V0__init_db.sql [hunk 4]
```diff
@@ -0,0 +1,83 @@
+
```
Comment (mibrasap) [inline services/upload/src/test/java/V0__init_db.sql:1]: As discussed, check if it's still required

File: services/upload/src/test/java/app/coronawarn/server/services/federation/upload/utils/MockData.java [hunk 5]
```diff
@@ -33,15 +36,27 @@ public static FederationUploadKey generateRandomUploadKey(boolean consentToShare
         .collect(Collectors.toList());
   }
 
+  public static int makeRollingStartIntervalFromSubmission(long submissionTimestamp) {
+    return (int)((submissionTimestamp) * 6);
+  }
+
   public static DiagnosisKey generateRandomDiagnosisKey(boolean consentToShare) {
```
Comment (mibrasap) [inline services/upload/src/test/java/app/coronawarn/server/services/federation/upload/utils/MockData.java:43]: As discussed, re-check recommended

File: services/upload/src/test/resources/application.yaml [hunk 6]
```diff
@@ -36,15 +36,19 @@ services:
...
     driver-class-name: org.testcontainers.jdbc.ContainerDatabaseDriver
-    url:  jdbc:tc:postgresql:11.8:///databasename?TC_TMPFS=/testtmpfs:rw
+    url: jdbc:tc:postgresql:11.8:///databasename?TC_TMPFS=/testtmpfs:rw?TC_INITSCRIPT=file:src/test/java/V0__init_db.sql
   test:
     database:
       # Use datasource as defined above.
       replace: none
-  main:
-    banner-mode: off
+  jpa:
```
Comment (mibrasap) [inline services/upload/src/test/resources/application.yaml:52]: Check if JPA can be removed

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/integration/S3PublisherTestIT.java [hunk 7]
```diff
@@ -47,6 +47,17 @@
...
+  }
+
+  @AfterEach
+  public void teardown() {
+    objectStoreAccess.deleteObjectsWithPrefix("");
```
Comment (hilmarf) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/integration/S3PublisherTestIT.java:58]: just my personal opinion on 'clean up after test' - what if something fails? How can you easily figure out, what went wrong, if all has been deleted AFTER the test execution? Of course to start always with a clean environment and a deterministic test setup, it makes sense to delete leftover BEFORE the test. But everything, which was created during the test execution, should be kept till the next run - shouldn't it?

**Review by hilmarf — DISMISSED** [review]

**Review by hilmarf — DISMISSED** [review]

**Review by mibrasap — DISMISSED** [review]

**Review by mibrasap — APPROVED** [review]

**Review by fredrb — APPROVED** [review]

**Review by sorin-iovita — APPROVED** [review]

**Pull request merged**